### PR TITLE
Code for updating invoice headers and footers

### PR DESCRIPTION
**Pull request by reybango**
Fixes # .

**Changes proposed in this pull request:**

Adding functionality to administration section to allow users to update invoice headers and footers. 
## 

cc @HospitalRun/core-maintainers

### CODE REVIEW

**Comment by reybango** [comment]
@taras here's the PR for the invoice header work.

File: app/admin/invoice-template/route.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import injectScript from 'ember-inject-script';
+
+export default AbstractEditRoute.extend({
+  hideNewButton: true,
+  newTitle: 'Invoice Template',
+  editTitle: 'Invoice Template',
+  
+  model: function() {
+    return new Ember.RSVP.Promise(function(resolve) {
+      this.get('store').find('option', 'invoice_template').then(function(invoiceOptions) {
```
Comment (billybonks) [inline app/admin/invoice-template/route.js]: can you use the es6 `=>` for keeping the correct `this` scope

File: app/admin/invoice-template/route.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import AbstractEditRoute from 'hospitalrun/routes/abstract-edit-route';
+import Ember from 'ember';
+import injectScript from 'ember-inject-script';
+
+export default AbstractEditRoute.extend({
+  hideNewButton: true,
+  newTitle: 'Invoice Template',
+  editTitle: 'Invoice Template',
+  
+  model: function() {
```
Comment (billybonks) [inline app/admin/invoice-template/route.js]: es6 `model()`

File: app/admin/invoice-template/route.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import AbstractEditRoute from 'hospitalrun/routes/abstract-edit-route';
+import Ember from 'ember';
+import injectScript from 'ember-inject-script';
+
+export default AbstractEditRoute.extend({
+  hideNewButton: true,
+  newTitle: 'Invoice Template',
```
Comment (billybonks) [inline app/admin/invoice-template/route.js]: `i18n`

File: app/admin/invoice-template/route.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  editTitle: 'Invoice Template',
```
Comment (billybonks) [inline app/admin/invoice-template/route.js]: `i18n`

File: app/admin/invoice-template/template.hbs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      <div role="tabpanel" class="tab-pane active" id="edit">
+        <div class="panel detail-section">
+          <div class="panel-heading">
+            <h3 class="panel-title">
+              Invoice Template
+            </h3>
+          </div>
+          <div class="panel-body detail-section-content">
+            <div class="row">      
+              <h4>Header Template</h4>
```
Comment (billybonks) [inline app/admin/invoice-template/template.hbs]: `i18n`

File: app/admin/invoice-template/template.hbs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      <li role="presentation" class="active"><a class="active" href="#edit" role="tab" data-toggle="pill" data-test-selector="general-tab">Edit</a></li>
+      <li role="presentation"><a class="active" href="#preview" role="tab" data-toggle="pill" data-test-selector="general-tab">Preview</a></li>
+    </ul>
+      
+    <div class="tab-content">
+      <div role="tabpanel" class="tab-pane active" id="edit">
+        <div class="panel detail-section">
+          <div class="panel-heading">
+            <h3 class="panel-title">
+              Invoice Template
```
Comment (billybonks) [inline app/admin/invoice-template/template.hbs]: `i18n`

File: app/admin/invoice-template/template.hbs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            </h3>
+          </div>
+          <div class="panel-body detail-section-content">
+            <div class="row">      
+              <h4>Header Template</h4>
+              {{textarea value=model.value.invoiceHeader}}
+              
+              <h4>Footer Template</h4>
```
Comment (billybonks) [inline app/admin/invoice-template/template.hbs]: `i18n`

File: app/admin/invoice-template/template.hbs [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      <div role="tabpanel" class="tab-pane" id="preview">
+        <div class="panel detail-section">
+          <div class="panel-heading">
+            <h3 class="panel-title">
+              Invoice Template Preview
+            </h3>
+          </div>
+          <div class="panel-body detail-section-content">
+            <div class="row">      
+              <h4>Header Template</h4>
```
Comment (billybonks) [inline app/admin/invoice-template/template.hbs]: `i18n`

File: app/admin/invoice-template/template.hbs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+              {{dynamic-render model.value.invoiceHeader}}
+              
+              <h4>Footer Template</h4>
```
Comment (billybonks) [inline app/admin/invoice-template/template.hbs]: `i18n`

File: app/admin/invoice-template/template.hbs [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+              {{textarea value=model.value.invoiceFooter}}
+            </div>
+          </div>
+        </div>
+      </div>
+      <div role="tabpanel" class="tab-pane" id="preview">
+        <div class="panel detail-section">
+          <div class="panel-heading">
+            <h3 class="panel-title">
+              Invoice Template Preview
```
Comment (billybonks) [inline app/admin/invoice-template/template.hbs]: `i18n`

File: app/mixins/navigation.js [hunk 11]
```diff
@@ -273,6 +273,12 @@ export default Ember.Mixin.create({
           iconClass: 'octicon-chevron-right',
           route: 'admin.roles',
           capability: 'user_roles'
+        },
+        {
+          title: 'Invoice Template',
```
Comment (billybonks) [inline app/mixins/navigation.js]: `i18n`

**Comment by taras** [comment]
@billybonks this PR needs some UI/UX attention. We should pull @jglovier in to look at this and then we can start doing clean up. Adding `i18n` and such.

**Comment by jglovier** [comment]
Can someone please post some screenshots of the UI changes in this PR?

**Comment by taras** [comment]
This PR is for issue #370

**Comment by taras** [comment]
# Screenshots

Here is what the new screens look like at the moment. We need to make them clearer.

## Administration -> Invoice Template -> Edit

screen shot 2016-07-17 at 1 06 40 pm

## Administration -> Invoice Template -> Preview

screen shot 2016-07-17 at 1 07 47 pm


# TODO 

- [ ] i18n the template strings
- [ ] [UX/UI] Need design for Administration -> Invoice Templates
- [ ] [UX/UI] Design/content for default templates 
- [ ] Pre-populate the database with default template

/cc @jglovier

**Comment by jkleinsc** [comment]
@taras thanks for picking this back up and updating this PR!  Looking forward to having Chris and Lennex contributing!

**Comment by tangollama** [comment]
Closing b/c this is covered by other features.

**Pull request closed without merging**